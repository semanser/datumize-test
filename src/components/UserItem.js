import React, { Component } from 'react'
import AvatarImage from './AvatarImage'

class UserItem extends Component {
  constructor(props) {
    super(props)

    this.state = { hovered: false }

    this.onClickHandler = this.onClickHandler.bind(this)
    this.onMouseEnter = this.onMouseEnter.bind(this)
    this.onMouseLeave = this.onMouseLeave.bind(this)
  }

  onMouseEnter() {
    this.setState({ hovered: true })
  }

  onMouseLeave() {
    this.setState({ hovered: false })
  }

  onClickHandler() {
    if (this.props.onSelect) {
      this.props.onSelect()
    }
  }

  render() {
    let button
    if (this.props.onUnassign) {
      button = this.state.hovered
        ? <div className="role unassign" onClick={this.props.onUnassign}>Unassign</div>
        : <div className="role">{this.props.user.role}</div>
    } 

    return (
      <div
        className="roles-item"
        onClick={this.onClickHandler}
        onMouseEnter={this.onMouseEnter}
        onMouseLeave={this.onMouseLeave}
      >
        <AvatarImage type="user" name={this.props.user.name}/>
        <div className="text">{this.props.user.name}</div>
        {button}
      </div>
    )
  }
}

export default UserItem

