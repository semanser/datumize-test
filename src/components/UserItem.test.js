import React from 'react'
import ReactDOM from 'react-dom'
import ShallowRenderer from 'react-test-renderer/shallow'
import UserItem from './UserItem'

const renderer = new ShallowRenderer()

describe('UserItem Component', () => {
  it('Renders component without crashing', () => {
    const div = document.createElement('div')
    const props = {
      user: {
        id: 1,
        name: 'John Doe',
        role: 'Admin',
      }
    }

    const tree = renderer.render(<UserItem {...props}/>)
    expect(tree).toMatchSnapshot()
  })

  it('should render the Unassign button if props.onUnassign is passed', () => {
    const div = document.createElement('div')
    const props = {
      user: {
        id: 1,
        name: 'John Doe',
        role: 'Admin',
      },
      onUnassign: () => {},
    }

    const tree = renderer.render(<UserItem {...props}/>)
    expect(tree).toMatchSnapshot()
  })
})

