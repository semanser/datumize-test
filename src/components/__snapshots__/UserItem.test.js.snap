// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem Component Renders component without crashing 1`] = `
<div
  className="roles-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <AvatarImage
    name="John Doe"
    type="user"
  />
  <div
    className="text"
  >
    John Doe
  </div>
</div>
`;

exports[`UserItem Component should render the Unassign button if props.onUnassign is passed 1`] = `
<div
  className="roles-item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <AvatarImage
    name="John Doe"
    type="user"
  />
  <div
    className="text"
  >
    John Doe
  </div>
  <div
    className="role"
  >
    Admin
  </div>
</div>
`;
